modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/exec/rel/RootNode.java

```diff
@@ -236,61 +214,55 @@ else if (state == State.END || state == State.CLOSED)
     }
 
     /** */
-    private Row take() {
+    private void exchangeBuffers() {
         assert !F.isEmpty(sources()) && sources().size() == 1;
 
         lock.lock();
         try {
             while (true) {
-                checkCancelled();
-                assert state == State.RUNNING;
+                //checkException();
```

Comment:
we need this call because without it client code will finish successfully in case it was waiting on a condition at the cancellation time

Reply:
return checkException